packages/interactivity/src/test/store.ts

```diff
@@ -370,4 +376,72 @@ describe( 'Interactivity API', () => {
 			} );
 		} );
 	} );
+
+	describe( 'getServerState', () => {
+		describe( 'should return a read-only generic object when no type is passed', () => {
+			// eslint-disable-next-line no-unused-expressions
+			() => {
+				const state = getServerState();
+				// @ts-expect-error
+				state.nonModifiable = 'error';
```

Comment:
The test description states that it "should return a read-only generic object when no type is passed." However, the type of `state` is just `object`, which has no properties, and the error returned is this one:

```
Property 'nonModifiable' does not exist on type 'object'.ts(2339)
```

Is this the expected output from TypeScript?

Reply:
The thing is that `Readonly< object >` -> `object`.

We could use `Readonly< Record< string, unknown >>`, but [it has a different behavior](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBCBGArAXDASgUwIYBNwBsBPAHgUQ2CgD4YBeGAbwF8BuAKDIDoxwBZEHAEsAZoKzx8GOjADkADxns2oSLDIAmVJlwESmUACccxaAcFgA5gBoYAVzABrHgHcwVGvWbsN3PgJFiElL08opAA):

- `object` type: `state.prop` is invalid, so mutating `state.prop` also fails.
- `Record< string, unknown >` type: `state.prop` is valid, but mutating `state.prop` fails.

It depends on what we want to do. If we want to force people who use TypeScript to have to type the state (`object`), or if we want to leave it more relaxed (`Record`).

I'm going to change to `Record` for now.